### ISSUE TITLE
Make the website a PWA (Progressive Web App)

### ISSUE THREAD

**Issue by nonsavant** [weg-li/weg-li#775]
Currently, if I save the website to my home screen in iOS, it's just a bookmark that opens in my browser.

If the app was a PWA, it would behave like a native app.

**Comment by phoet** [comment]
PWA are dead, see related issues

**Comment by nonsavant** [comment]
The only vaguely related issue I can find is https://github.com/weg-li/weg-li/issues/514, which is talking about getting an app into Google play store. 

Seems like making a website a PWA is relatively simple (manifest + service worker) and seeing as there is no prospect of wegli being available as a Store App any time soon, seems like the next best alternative.

**Comment by phoet** [comment]
Discussion was on Twitter https://x.com/weg_li/status/1788308125713338403

**Comment by GoetzGoerisch** [comment]
May I add my two cents.
PWAs are not dead and Apple back paddled on this regard.

Could you please bring this discussion to this thread, as twitter or x is an not inclusive platform, many are not using and cannot view?

What are the arguments against it?

**Comment by phoet** [comment]
It was dead at the time of discussion, what are the relevant news regarding the back paddling?

I've cleaned my memory about PWA when that was announced, but I think it just had minimal advantages compared to a simple bookmark.

**Comment by GoetzGoerisch** [comment]
There was this uproar after Apple announced that due to DMA they would kill PWA.
But the [back paddled shortly after](https://heise.de/-9644176).

If a PWA is now more or less usefull than a bookmark is a different story.
Implementing or not as well.

Simply posting "PWA are dead, see related issues" without providing relevant pointers to issues and simply closing this issue is not really nice.

**Comment by GoetzGoerisch** [comment]
From the Apple Developer guide:

https://developer.apple.com/support/dma-and-apps-in-the-eu/#dev-qa

> UPDATE: Previously, Apple announced plans to remove the Home Screen web apps capability in the EU as part of our efforts to comply with the DMA. The need to remove the capability was informed by the complex security and privacy concerns associated with web apps to support alternative browser engines that would require building a new integration architecture that does not currently exist in iOS.

>We have received requests to continue to offer support for Home Screen web apps in iOS, therefore we will continue to offer the existing Home Screen web apps capability in the EU. This support means Home Screen web apps continue to be built directly on WebKit and its security architecture, and align with the security and privacy model for native apps on iOS.

> Developers and users who may have been impacted by the removal of Home Screen web apps in the beta release of iOS in the EU can expect the return of the existing functionality for Home Screen web apps with the availability of iOS 17.4 in early March.

**Comment by nonsavant** [comment]
Yes, PWAs are still very much alive. I just added one (recipesage.com) on iOS 17.5.1 and it worked as expected (which is what lead me to try the same with wegli, only to discover that in wegli’s case, it just saves a bookmark). Works on Android too. 

Advantages? The end user effectively has an app that behaves to all extents and purposes like a native app installed from an official store. Doesn’t open in the browser anymore, the browser controls are not shown, more screen real estate. But push notifications can also be supported. 

As I wrote in my initial post, wegli could also support this behaviour and it seems like a relatively easy win. 

Even if this can’t be implemented any time soon, I think the issue is still valid and could be reopened.

**Comment by phoet** [comment]
We wont be releasing anything from the official apple store, so there is that. But since Apple is being forced to open side-loading there is at least the possibility of providing support for stuff like that.

**Comment by rudynz** [comment]
I made an Android PWA with the help of clade code to be able to share photos directly from the gallery via the share function. It uses the weg.li API. The code can be found here:
https://github.com/rudynz/weg.li-pwa

Test instances for main and dev branch here:
https://wegli.netlify.app/
https://dev--wegli.netlify.app/

Let me know if you find this a helpful approach.

**Comment by phoet** [comment]
mehr als "the edge function timed out" konnte ich dem nicht entlocken

**Comment by rudynz** [comment]
> mehr als "the edge function timed out" konnte ich dem nicht entlocken

kannst du etwas mehr sagen was du versucht hast, selber bauen, einen der branches auf netlify? Welcher Browser, welches OS?
